cmd/podman/images/scp.go

```diff
@@ -379,3 +426,119 @@ func parseArgs(args []string, cfg *config.Config) (map[string]config.Destination
 	}
 	return serv, nil
 }
+
+// execLoad executes the podman load command given the podman binary
+func execLoad(podman string) error {
+	var loadCommand []string
+	if scpOpts.Save.Quiet || scpOpts.Load.Quiet {
+		loadCommand = []string{"load", "-q", "--input", scpOpts.Save.Output}
+	} else {
+		loadCommand = []string{"load", "--input", scpOpts.Save.Output}
+	}
+	if rootless.IsRootless() {
+		cmd := exec.Command(podman)
+		cmd.Args = append(cmd.Args, parentFlags...)
+		cmd.Args = append(cmd.Args, loadCommand...)
+		cmd.Env = os.Environ()
+		cmd.Stderr = os.Stderr
+		cmd.Stdout = os.Stdout
+		logrus.Debug("Executing load command")
+		err := cmd.Run()
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	machinectl, err := exec.LookPath("machinectl")
+	if err != nil {
+		cmd := exec.Command("su", "-l", "root", "--command")
```

Comment:
I think this should only be done with sudo never with su.

Reply:
Ok as long as this is only done with Podman is run as root.

Reply:
Yeah, this only happens if rootless.IsRootless() is false, otherwise we use a `sudo podman` combined with regular `podman`

Reply:
Can you break the rootless code into a separate function to make this read easier?  I still have a hard time reading this code and knowing that the process is running as root.